### PR TITLE
Lawyer badge changes typing indicator

### PR DESCRIPTION
**Pull request by Morb0**
## About the PR

Adds a component `TypingIndicatorClothing` for clothing, which changes the indicator of the person wearing the thing.
Prototype resolving is moved to `OnAppearanceChange` since the `Protype` field can now change.
Add new type of typing indicator (sprites already exist).

In shared system added const with `InitialIndicatorId`, it used as default id when the item is stripped.
So potentially if 2 items are worn and one of them is stripped, the indicator will not use the prototype of the remaining item worn and will be just reset to default.

**Media**

image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Worn lawyer badge now changes the style of the typing indicator

### CODE REVIEW

File: Content.Shared/Chat/TypingIndicator/TypingIndicatorClothingComponent.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Shared.Chat.TypingIndicator;
+
+[RegisterComponent]
+public sealed class TypingIndicatorClothingComponent : Component
+{
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("proto", required: true, customTypeSerializer: typeof(PrototypeIdSerializer<TypingIndicatorPrototype>))]
+    public string Prototype = default!;
```
Comment (metalgearsloth) [inline Content.Shared/Chat/TypingIndicator/TypingIndicatorClothingComponent.cs:10]: This isn't networked? Also needs Access.

File: Content.Client/Chat/TypingIndicator/TypingIndicatorVisualizerSystem.cs [hunk 2]
```diff
@@ -9,42 +9,28 @@ public sealed class TypingIndicatorVisualizerSystem : VisualizerSystem<TypingInd
...
+        
         if (!_prototypeManager.TryIndex<TypingIndicatorPrototype>(component.Prototype, out var proto))
         {
             Logger.Error($"Unknown typing indicator id: {component.Prototype}");
             return;
         }
 
-        var layer = sprite.LayerMapReserveBlank(TypingIndicatorLayers.Base);
+        args.Component.TryGetData(TypingIndicatorVisuals.IsTyping, out bool isTyping);
+        var isLayerExist = sprite.LayerMapTryGet(TypingIndicatorLayers.Base, out var layer);
```
Comment (metalgearsloth) [inline Content.Client/Chat/TypingIndicator/TypingIndicatorVisualizerSystem.cs:26]: `layerExists` or `doesLayerExist`

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Client/Chat/TypingIndicator/TypingIndicatorVisualizerSystem.cs [hunk 3]
```diff
@@ -9,42 +9,28 @@ public sealed class TypingIndicatorVisualizerSystem : VisualizerSystem<TypingInd
...
     {
-        base.Initialize();
-        SubscribeLocalEvent<TypingIndicatorComponent, ComponentInit>(OnInit);
-    }
+        base.OnAppearanceChange(uid, component, ref args);
 
-    private void OnInit(EntityUid uid, TypingIndicatorComponent component, ComponentInit args)
-    {
         if (!TryComp(uid, out SpriteComponent? sprite))
             return;
```
Comment (metalgearsloth) [inline Content.Client/Chat/TypingIndicator/TypingIndicatorVisualizerSystem.cs:17]: AppearanceChangeEvent already has a reference to spritecomp I think.

Comment (Morb0) [reply Content.Client/Chat/TypingIndicator/TypingIndicatorVisualizerSystem.cs]: Yep, done

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**